jerry-core/ecma/operations/ecma-proxy-object.c

```diff
@@ -310,8 +310,89 @@ ecma_value_t
 ecma_proxy_object_is_extensible (ecma_object_t *obj_p) /**< proxy object */
 {
   JERRY_ASSERT (ECMA_OBJECT_IS_PROXY (obj_p));
-  JERRY_UNUSED (obj_p);
-  return ecma_raise_type_error (ECMA_ERR_MSG ("UNIMPLEMENTED: Proxy.[[IsExtensible]]"));
+
+  ecma_proxy_object_t *proxy_obj_p = (ecma_proxy_object_t *) obj_p;
+
+  /* 1. */
+  ecma_value_t handler = proxy_obj_p->handler;
+
+  /* 2. */
+  if (ecma_is_value_null (handler))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Handler can not be null."));
+  }
+
+  /* 3. */
+  JERRY_ASSERT (ecma_is_value_object (handler));
+
+  /* 4. */
+  ecma_value_t target = proxy_obj_p->target;
+
+  /* 5. */
+  ecma_value_t trap = ecma_op_get_method_by_magic_id (handler, LIT_MAGIC_STRING_IS_EXTENSIBLE);
+
+  /* 6. */
+  if (ECMA_IS_VALUE_ERROR (trap))
+  {
+    return trap;
+  }
+
+  ecma_object_t *target_obj_p = ecma_get_object_from_value (target);
+
+  /* 7. */
+  if (ecma_is_value_undefined (trap))
+  {
+    if (ECMA_OBJECT_IS_PROXY (target_obj_p))
```

Comment:
It would be better to use the "general" method for is extensible here: `ecma_builtin_object_object_is_extensible`

Reply:
+1 for this idea.